src/storage/src/table/mem_table.rs

```diff
@@ -45,26 +59,62 @@ impl MemTable {
     }
 
     /// read methods
-    pub async fn get_row(&self, pk: &Row) -> StorageResult<Option<RowOp>> {
-        todo!()
+    pub fn get_row(&self, pk: &Row) -> StorageResult<Option<RowOp>> {
+        let res = self.buffer.get(pk);
+        match res {
+            Some(row_op) => Ok(Some(row_op.clone())),
+            None => Ok(None),
+        }
     }
 
     /// write methods
-    pub async fn insert(&mut self, _pk: Row, _value: Row) -> StorageResult<()> {
+    pub fn insert(&mut self, pk: Row, value: Row) -> StorageResult<()> {
+        let entry = self.buffer.entry(pk);
+        match entry {
+            Entry::Vacant(e) => {
+                e.insert(RowOp::Insert(value));
+            }
+            Entry::Occupied(mut e) => match e.get() {
+                RowOp::Delete(_) => {
+                    e.insert(RowOp::Update(value));
+                }
+                _ => {
+                    panic!(
+                        "invalid flush status: double insert {:?} -> {:?}",
+                        e.key(),
+                        value
+                    );
+                }
+            },
+        }
         Ok(())
     }
 
-    pub async fn delete(&mut self, _pk: Row, _old_value: Row) -> StorageResult<()> {
+    pub fn delete(&mut self, pk: Row, old_value: Row) -> StorageResult<()> {
+        let entry = self.buffer.entry(pk);
+        match entry {
+            Entry::Vacant(e) => {
+                e.insert(RowOp::Delete(old_value));
+            }
+            Entry::Occupied(mut e) => {
+                if e.get().is_insert() {
+                    e.remove();
+                } else if e.get().is_update() {
+                    e.insert(RowOp::Update(old_value));
+                } else {
+                    panic!(
```

Comment:
I think maybe it should be match here... btw should we maintain the `is_insert` and `is_update` method? now our enum has express them clearly and the wrapping seems not needed.